lib/services/github.go

```diff
@@ -183,6 +183,16 @@ func (c *GithubConnectorV3) GetMetadata() Metadata {
 	return c.Metadata
 }
 
+// WithoutSecrets returns an instance of resource without secrets.
+func (c *GithubConnectorV3) WithoutSecrets() Resource {
+	if c.GetClientSecret() == "" {
+		return c
+	}
+	c2 := *c
```

Comment:
This won't deep-copy `GithubConnectorSpecV3.TeamsToLogins`, is that expected?
To be on the safe side, consider adding `Clone` methods for these types

Reply:
I considered having `WithoutSecrets` perform a deep copy, but I ended up liking the idea of having this be a no-op if no secrets were present more (and only performing a deep copy sometimes is worse than doing it never imo).